3scale/porta
# [WIP] Run CI test-suite locally, in containers

**What this PR does / why we need it**:

This PR aims to do 2 things: 

* Give developers the chance to run the tests on their own machine (through a docker-compose set up that orchestrates the necessary containers required for running the tests). It should be noted that one can do so, **without needing to install any of this project's dependencies on the host system**, since dependencies are only installed in the containers and in local volumes. This should lower the entry barrier for both external contributors, as well as new members on the team. 

* Move test commands to make targets, rather than including them in the circleci config directly, so that they can be reused in the dev environment. At the same time this makes it easier to move to a different CI, if for any reason the need comes up... All we would need to do is have the CI server spin up a set of containers and invoke the make targets in the same way they are invoked in circleci. This was not as easy to do when we needed to provision VMs - now we just need containers. 

**Which issue(s) this PR fixes** 

Fixes #93 


**Verification steps** 

Once this PR is complete, an external contributor should be able to run `make test` (or similar) that will run the same tests that would run on circleci, but instead run them on the contributor's machine. 


**Special notes for your reviewer**:

N/A
